### ISSUE TITLE
Hangs on Install

### ISSUE THREAD

**Issue by JasonTheIncredible** [date-fns/date-fns#2479]
I'm trying to install date-fns for the first time. I tried using the suggested `npm install date-fns --save` as well as `npm install date-fns`. I tried manually adding it to my package.json dependencies `"date-fns": "^2.21.3"` then running `npm update` but it always freezes at `reify:date-fns: sill audit bulk request` (see screenshot). I've let it sit like that for an hour before with no progress.

![date-fns](https://user-images.githubusercontent.com/35417691/118612310-428bf780-b783-11eb-96b1-1ee19b2d97c7.png)

Any idea what might be causing this?

Edit: I've also tried manually installing an older version `"date-fns": "2.20.3"` with the same results.

**Comment by fturmel** [comment]
I've never used `reify` before, but I quickly ran `npm init -y && npm install --save reify date-fns --verbose` in an empty directory and it went through without issues. Your terminal message looks different than mine though, maybe you need to bump your version of that library?

```bash
[...]
npm timing reify:audit Completed in 144ms
npm timing reifyNode:node_modules/reify Completed in 184ms
npm timing reifyNode:node_modules/date-fns Completed in 607ms
npm timing reify:unpack Completed in 607ms
npm timing reify:unretire Completed in 0ms
[...]
```

<details>
<summary>Detailed terminal output...</summary>

```sh
Wrote to /Users/fturmel/Documents/projects/reify-demo/package.json:

{
  "name": "reify-demo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}


npm verb cli [
npm verb cli   '/opt/homebrew/Cellar/node/16.2.0/bin/node',
npm verb cli   '/opt/homebrew/bin/npm',
npm verb cli   'install',
npm verb cli   '--save',
npm verb cli   'reify',
npm verb cli   'date-fns',
npm verb cli   '--verbose'
npm verb cli ]
npm info using npm@7.14.0
npm info using node@v16.2.0
npm timing npm:load:whichnode Completed in 1ms
npm timing config:load:defaults Completed in 0ms
npm timing config:load:file:/opt/homebrew/lib/node_modules/npm/npmrc Completed in 1ms
npm timing config:load:builtin Completed in 1ms
npm timing config:load:cli Completed in 1ms
npm timing config:load:env Completed in 0ms
npm timing config:load:file:/Users/fturmel/Documents/projects/reify-demo/.npmrc Completed in 0ms
npm timing config:load:project Completed in 0ms
npm timing config:load:file:/Users/fturmel/.npmrc Completed in 0ms
npm timing config:load:user Completed in 0ms
npm timing config:load:file:/opt/homebrew/etc/npmrc Completed in 0ms
npm timing config:load:global Completed in 0ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:credentials Completed in 1ms
npm timing config:load:setEnvs Completed in 0ms
npm timing config:load Completed in 3ms
npm timing npm:load:configload Completed in 3ms
npm timing npm:load:setTitle Completed in 2ms
npm timing npm:load:setupLog Completed in 1ms
npm timing npm:load:cleanupLog Completed in 1ms
npm timing npm:load:configScope Completed in 0ms
npm timing npm:load:projectScope Completed in 0ms
npm timing npm:load Completed in 9ms
npm timing config:load:flatten Completed in 1ms
npm timing arborist:ctor Completed in 1ms
npm timing arborist:ctor Completed in 0ms
npm timing idealTree:init Completed in 6ms
npm timing idealTree:userRequests Completed in 1ms
npm http fetch GET 304 https://registry.npmjs.org/reify 529ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/date-fns 52ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/acorn-dynamic-import 36ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/magic-string 99ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/acorn 108ms (from cache)
npm http fetch GET 304 https://registry.npmjs.org/semver 110ms (from cache)
npm timing idealTree:#root Completed in 708ms
npm timing idealTree:node_modules/date-fns Completed in 0ms
npm http fetch GET 304 https://registry.npmjs.org/sourcemap-codec 36ms (from cache)
npm timing idealTree:node_modules/reify Completed in 43ms
npm timing idealTree:node_modules/acorn Completed in 1ms
npm timing idealTree:node_modules/acorn-dynamic-import Completed in 0ms
npm timing idealTree:node_modules/magic-string Completed in 1ms
npm timing idealTree:node_modules/semver Completed in 0ms
npm timing idealTree:node_modules/sourcemap-codec Completed in 0ms
npm timing idealTree:buildDeps Completed in 753ms
npm timing idealTree:fixDepFlags Completed in 1ms
npm timing idealTree Completed in 761ms
npm timing reify:loadTrees Completed in 761ms
npm timing reify:diffTrees Completed in 1ms
npm timing reify:retireShallow Completed in 0ms
npm timing reify:createSparse Completed in 1ms
npm timing reify:loadBundles Completed in 0ms
npm timing reifyNode:node_modules/acorn-dynamic-import Completed in 106ms
npm timing reifyNode:node_modules/semver Completed in 106ms
npm timing reifyNode:node_modules/sourcemap-codec Completed in 107ms
npm timing reifyNode:node_modules/magic-string Completed in 109ms
npm timing reifyNode:node_modules/acorn Completed in 115ms
npm http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 143ms
npm timing auditReport:getReport Completed in 144ms
npm timing auditReport:init Completed in 0ms
npm timing reify:audit Completed in 144ms
npm timing reifyNode:node_modules/reify Completed in 184ms
npm timing reifyNode:node_modules/date-fns Completed in 607ms
npm timing reify:unpack Completed in 607ms
npm timing reify:unretire Completed in 0ms
npm timing build:queue Completed in 1ms
npm timing build:link:node_modules/semver Completed in 1ms
npm timing build:link:node_modules/acorn Completed in 1ms
npm timing build:link Completed in 1ms
npm timing build:deps Completed in 2ms
npm timing build Completed in 2ms
npm timing reify:build Completed in 2ms
npm timing reify:trash Completed in 0ms
npm timing reify:save Completed in 2ms
npm timing reify Completed in 1379ms

added 7 packages, and audited 8 packages in 1s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm timing command:install Completed in 1381ms
npm verb exit 0
npm timing npm Completed in 1464ms
npm info ok
```
</details>  


I doubt this is a `date-fns` issue but happy to help if you still are experiencing issues. Please share your package.json file and a minimal code example to reproduce.

**Comment by JasonTheIncredible** [comment]
@rikkalo

Hi @fturmel. Thanks for your response and sorry for my late response - I wasn't at my computer so I couldn't do any testing.

I ran `npm init -y && npm install --save date-fns` in an empty directory and it installed fine. However, installing something before date-fns caused date-fns to hang. Example:

In a new directory:
`npm init -y && npm install --save date-fns` - works great!

In a new directory:
`npm init -y && npm install --save firebase-admin && npm install --save firebase-functions` - works as expected.
`npm install --save date-fns` - hangs as described in the original post.

In a new directory:
`npm init -y && npm install --save needle` - works as expected.
`npm install --save date-fns` - hangs as described in the original post.

For good measure, here's my package.json file:
```
{
  "name": "functions",
  "description": "Cloud Functions for Firebase",
  "scripts": {
    "lint": "eslint .",
    "serve": "firebase emulators:start --only functions",
    "shell": "firebase functions:shell",
    "start": "npm run shell",
    "deploy": "firebase deploy --only functions",
    "logs": "firebase functions:log"
  },
  "engines": {
    "node": "12"
  },
  "main": "index.js",
  "dependencies": {
    "firebase-admin": "^9.2.0",
    "firebase-functions": "^3.11.0",
    "needle": "^2.6.0",
    "twilio": "^3.62.0"
  },
  "devDependencies": {
    "eslint": "^7.6.0",
    "eslint-config-google": "^0.14.0",
    "firebase-functions-test": "^0.2.0"
  },
  "private": true
}

**Comment by fturmel** [comment]
I can't reproduce the issue following your directions or starting from your package.json and adding date-fns. It certainly feels like a local npm problem to me. Have you been able to replicate this on other workstations or servers? Which version of node / npm are you using?

Couple things to try:
- Very obvious, but have you tried a fresh node install? Updating npm itself and global packages with `npm -g update`?
- I doubt this is the problem if date-fns installed in some scenarios, but maybe you could try clearing the npm cache in case something is corrupted? `npm cache clean --force`
- There is a command to check for general setup issues, maybe that can reveal something useful. `npm doctor`
- Can you pass the `--verbose` option to the install command so we can get more context around the step that gets stuck?
- Have you tried with npm alternatives like yarn or pnpm to compare?

**Comment by JasonTheIncredible** [comment]
I finally got it to install! Thank you so much @fturmel!

tl;dr: I updated everything (and cleaned the cache) and it worked.

For anyone with this problem in the future, here's the step-by-step guide of what I did:

- First I ran `npm -g update` and got the following message:

```
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

removed 410 packages, changed 714 packages, and audited 748 packages in 32s
```

- Then I ran `npm cache clean --force` without a problem.

- Then I ran `npm doctor` (which I didn't know was a thing but turned out to be very helpful) and got the following:
```
Check                    Value   Recommendation/Notes
npm ping                 ok
npm -v                   ok      current: v7.15.1, latest: v7.15.1
node -v                  not ok  Use node v14.17.0 (current: v12.18.0)
npm config get registry  ok      using default registry (https://registry.npmjs.org/)
which git                ok      C:\Program Files\Git\cmd\git.EXE
Verify cache contents    ok      verified 2 tarballs

npm ERR! Some problems found. See above for recommendations.
```
So I went to https://nodejs.org/en/ and updated to the LTS. I then tried running `npm doctor` again but got the error message:
```
Error: Cannot find module '@npmcli/config'
```
After doing some Googling, I came across the solution in https://stackoverflow.com/questions/24721903/npm-npm-cli-js-not-found-when-running-npm to go to "Apps and Features" in Windows 10, find Node.js, click "Modify", then click "Repair".

- Then I ran `npm doctor` again and got the following:
```
Check                    Value   Recommendation/Notes
npm ping                 ok
npm -v                   not ok  Use npm v7.15.1
node -v                  ok      current: v14.17.0, recommended: v14.17.0
npm config get registry  ok      using default registry (https://registry.npmjs.org/)
which git                ok      C:\Program Files\Git\cmd\git.EXE
Verify cache contents    ok      verified 2 tarballs

npm ERR! Some problems found. See above for recommendations.
```
- So I ran `npm install -g npm@latest` without a problem.

- Then I ran `npm doctor` again to get the following:

```
Check                    Value  Recommendation/Notes
npm ping                 ok
npm -v                   ok     current: v7.15.1, latest: v7.15.1
node -v                  ok     current: v14.17.0, recommended: v14.17.0
npm config get registry  ok     using default registry (https://registry.npmjs.org/)
which git                ok     C:\Program Files\Git\cmd\git.EXE
Verify cache contents    ok     verified 1271 tarballs
```

- I then ran `npm install date-fns` and it worked! Best of luck to everyone!

**Comment by fturmel** [comment]
Great! Thank you for sharing the details.

**Comment by muks2p** [comment]
Thanks @JasonTheIncredible You saved my day

**Comment by ChaimaSka** [comment]
I've installed date-fns, but I'm encountering issues during the build process of the program!Unfortunately, I'm unsure of the root cause of these problems please help
![Annotation](https://github.com/date-fns/date-fns/assets/143667515/fbe5b94e-9a4a-4622-b08b-3a7022a3efda)
